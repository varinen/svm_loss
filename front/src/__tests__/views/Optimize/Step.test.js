import React from 'react';
import {createShallow} from '@material-ui/core/test-utils';
import Step from '../../../views/SVMLoss/Demo/Optimize/Step';

describe('Step control', () => {
    let component, props, shallow;
    const mockToggleOptimize = jest.fn();
    const mockSetAvailableIterations = jest.fn();
    beforeEach(() => {
        props = {
            params: {},
            data: [],
            hyper: {},
            step: {iteration: 0, availableIterations: 0},
            toggleOptimize: mockToggleOptimize,
            setAvailableIterations: mockSetAvailableIterations
        };
        shallow = createShallow({dive: true});
        component = shallow(<Step {...props}/>);
    });

    it('renders correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Click on Single Step causes the toggleOptimize and setAvailableIterations be called', () => {
        const additional = [];
        component.find('#singe-step')
            .simulate('click');
        expect(mockToggleOptimize).toHaveBeenCalledWith(true);
        expect(mockSetAvailableIterations).toHaveBeenCalledWith(1);
    });
});
