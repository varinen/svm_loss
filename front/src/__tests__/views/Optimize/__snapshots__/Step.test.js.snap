// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step control renders correctly 1`] = `
<Fragment>
  <h4>
    Optimize
  </h4>
  <GridContainer
    className=""
    container={true}
    spacing={2}
  >
    <GridItem
      className=""
      key="single_step"
      xs={6}
    >
      <p>
        Iterations: 
        0
      </p>
      <ForwardRef
        color="primary"
        id="singe-step"
        onClick={[Function]}
      >
        Single Step
      </ForwardRef>
       
    </GridItem>
    <GridItem
      className=""
      key="auto_update"
      xs={6}
    >
      <p>
        Â 
      </p>
      <WithStyles(ForwardRef(FormControlLabel))
        control={
          <ForwardRef(WithStyles)
            checked={false}
            color="primary"
            onChange={[Function]}
          />
        }
        label="Run for 25 auto iterations"
      />
    </GridItem>
  </GridContainer>
</Fragment>
`;
